source/hp/dof_handler.cc

```diff
@@ -1731,13 +1731,11 @@ namespace hp
   {
     // connect functions to signals of the underlying triangulation
     tria_listeners.push_back(this->tria->signals.pre_refinement.connect(
-      std::bind(&DoFHandler<dim, spacedim>::pre_refinement_action,
-                std::ref(*this))));
+      [&]() { this->pre_refinement_action(); }));
```

Comment:
How is this actually implemented by compilers? Does the anonymous struct that is used to represent a lambda include *all* locally visible symbols, or just the ones actually referenced in the lambda itself?

I'm asking because this happens to be an object that has a long lifetime: Until the triangulation dies. If it stores references to all local variables, then that's a lot. I suspect that it doesn't, but it might be easier to just write `[this]` here.

Reply:
No, the compiler can figure out which variables are actually needed.

Reply:
And I suspect it does, right?

Reply:
I mean not only does the standard allow it, but the compilers actually do that. (In reality, I can't imagine anything else -- the objects would be exceptionally large otherwise...)

Reply:
Yes, but I agree that is better to be specific. Will change.